Documentation on System PCG for Self Hosted Palette Installation (#1348)
spectrocloud/librarium: content/docs/04-clusters/02-data-center/01-maas/10-register-manage-maas-cloud-accounts.md

# Prerequisites

> - An installed PCG if you do not have a direct connection to the MAAS environment. 

Reviewer: Add a link to the Install PCG page.

Author: Added

>   If are self-hosting Palette and have a direct connection to the MAAS environment, you can select **Use System Private Gateway**.

Reviewer 2: You might want to add a link to the architecture doc here. Something like: 

To learn more about  when you would use Palette's PCG or the System Private Gateway, refer to the table in the [Architecture](/clusters/data-center/maas/architecture) guide.

Author: Added Rita.

>   If are self-hosting Palette and have a direct connection to the MAAS environment, you can select **Use System Private Gateway**. To learn more about when you would use Palette's PCG or the System Private Gateway, refer to the table in the [Architecture](/clusters/data-center/maas/architecture) guide.

Reviewer: Let's not refer to the table but rather the PCG section you will add in the architecture page.

Author: Sure.

- An installed PCG. Review [Install and Manage MAAS Gateway](/clusters/data-center/maas/install-manage-maas-pcg) for guidance.

> - For Self Hosted Palette, ensure Palette has network access to MAAS server.

Reviewer 2: ```suggestion
- For self-hosted Palette, ensure Palette has network access to the MAAS server.
```

Refer to [Register a Self-Hosted Palette Maas Cloud Account](#register-a-self-hosted-palette-maas-cloud-account) to learn more.

- An active [MAAS API key](https://maas.io/docs/api-authentication-reference) which can be generated in the MAAS web console under **My Preferences** > **API keys**. The following is an example key:

  ``APn53wz232ZwBMxDp5:MHZIbUp3e4DJTjZEKg:mdEv33WAG536MhNC8mIywNLtjcDTnFAQ``

 For details, refer to the MAAS document on [how to add an API key](https://maas.io/docs/how-to-manage-user-accounts#heading--api-key).

> 

Reviewer: This could be confusing. A user might ask "How do I know if I need a PCG or not. How are we helping the user understand that in this document? 

Also, because the steps will be very similar, we can improve the reader experience by using Tabs. With PCG, and Without PCG.

Author: Fixed this according to what we discussed.

3. Locate **MAAS** on the **Cloud Accounts** page and click **Add MAAS Account**.

> 4. On the **Add maas cloud account** pop-up, enter values for properties listed in the following table.

Reviewer 2: ```suggestion
4. In the next window that displays, enter values for properties listed in the following table.
```

>     For a self-hosted environment, you can register a MAAS cloud account without installing a PCG. Toggle **Use System Private Gateway** to register without a PCG.

Reviewer 2: ```suggestion
    In a self-hosted environment where Palette has direct network access to MAAS, you can register a MAAS cloud account without installing the PCG. Note the **Use System Private Gateway** setting listed in the table.
```

>     In a self-hosted environment where Palette has direct network access to MAAS, you can register a MAAS cloud account without installing the PCG. Note the **Use System Private Gateway** setting listed in the table.

Reviewer: Add a link to the Architecture page where System CG is discussed.

Author: Added.

| Property | Description |
|-----------|-------------|
| Account Name | Custom name for the cloud name. |

> | Use System Private Gateway | Toggle this option to bypass PCG install. |

Reviewer 2: ```suggestion
| Use System Private Gateway | This setting is for self-hosted environments that do not require a PCG. Toggle this option to bypass installing the PCG.|
```

> | Select Private Cloud Gateway | Select your MAAS cloud gateway from the drop-down menu. |

Reviewer 2: ```suggestion
| Select Private Cloud Gateway | Select your MAAS cloud gateway from the **drop-down Menu**. |
```

> | API Endpoint | API Endpoint of the gateway. |

Reviewer 2: ```suggestion
| API Endpoint | API endpoint of the gateway. |
```

| API Endpoint | API Endpoint of the gateway.|
| API Key | The MAAS API key. |

> # Register a Self-Hosted Palette Maas Cloud Account

Author: Yes Rita, I did that in the PR.

If you're registering a Maas cloud account using a self-hosted Palette, ensure your self-hosted Palette has network access to the Maas server. 
Follow these steps to register a Maas cloud account using self-hosted Palette. 

<br />

1. Follow steps 1-3 listed in [Register a MAAS Cloud Account](#register-a-maas-cloud-account). 

2. Once you click **Add MAAS Account**, the **Add maas cloud account** pop-up appears. Enter your **Account name.**

3. Select the **Use System PCG** checkbox.

<InfoBox>

> The system PCG enables direct connectivity between Palette and MAAS since there is no firewall or NAT to traverse. Selecting the **Use System PCG** checkbox ensures you register a Maas cloud account without installing PCG.

Author: Sure I removed them from here.
